packages/core/src/shadow/PipelinePass.ts

```diff
@@ -0,0 +1,21 @@
+import { Engine } from "../Engine";
+import { CullingResults } from "../RenderPipeline/CullingResults";
+import { RenderContext } from "../RenderPipeline/RenderContext";
+
+/**
+ * PipelinePass is a base class for all pipeline passes.
+ */
+export abstract class PipelinePass {
+  protected _engine: Engine;
+
+  constructor(engine: Engine) {
```

Comment:
I think based on `Camera` is better, `DepthOnlyPass` and `CascadedShadowCasterPass` will be more suitable

Reply:
Different camera use the same Pipeline pass instance, should not bind camera in constrouctor